packages-ts/gauntlet-terra-contracts/src/commands/contracts/ocr2/setBilling.ts

```diff
@@ -33,13 +33,42 @@ const makeContractInput = async (input: CommandInput): Promise<ContractInput> =>
     config: {
       observation_payment_gjuels: new BN(input.observationPaymentGjuels).toNumber(),
       transmission_payment_gjuels: new BN(input.transmissionPaymentGjuels).toNumber(),
-      recommended_gas_price_uluna: new BN(input.recommendedGasPriceUluna).toString(),
+      recommended_gas_price_uluna: input.recommendedGasPriceUluna
     },
   }
 }
 
-// TODO: Add validation
 const validateInput = (input: CommandInput): boolean => {
+  let observationPayment:BN
+  let transmissionPayment:BN
+  let gas_price:number = Number(input.recommendedGasPriceUluna) // parse as float64
+  if (!isFinite(gas_price)) {
+    throw new Error(
+      `recommendedGasPriceUluna=${input.recommendedGasPriceUluna} is not a valid floating point number.`
+    )
+  }
+
+  if (gas_price < 0.0) {
+    throw new Error(`recommendedGasPriceUluna=${input.recommendedGasPriceUluna} cannot be negative`)
+  }
+
+  try {
+      observationPayment = BN(input.observationPaymentGjuels)
+      transmissionPayment = BN(input.transmissionPaymentGjuels) // parse as integers
+      if (observationPayment.isNeg() ||
+             transmissionPayment.isNeg()
+         ) {
+        throw new Error(
+          `observationPaymentGjuels=${input.observationPaymentGjuels} and ` +
+          `transmissionPaymentGjuels=${input.transmissionPaymentGjuels} cannot be negative`
+        )
+      }
+  } catch {
+      throw new Error(
+          `observationPaymentGjuels=${input.observationPaymentGjuels} and ` +
+          `transmissionPaymentGjuels=${input.transmissionPaymentGjuels} must both be integers`
+      )
+  }
```

Comment:
Let's avoid nesting if possible for readability. This could be:
```suggestion
 try {
    observationPayment = BN(input.observationPaymentGjuels)
    transmissionPayment = BN(input.transmissionPaymentGjuels) // parse as integers
  } catch {
    throw new Error(
      `observationPaymentGjuels=${input.observationPaymentGjuels} and ` +
        `transmissionPaymentGjuels=${input.transmissionPaymentGjuels} must both be integers`,
    )
  }
  if (observationPayment.isNeg() || transmissionPayment.isNeg()) {
    throw new Error(
      `observationPaymentGjuels=${input.observationPaymentGjuels} and ` +
        `transmissionPaymentGjuels=${input.transmissionPaymentGjuels} cannot be negative`,
    )
  }
```